### PR TITLE
[JENKINS-70172] Use project configure permission for reset quality gate action

### PR DESCRIPTION
**Pull request by nitschSB**
[JENKINS-70172](https://issues.jenkins.io/browse/JENKINS-70172)

Currently only users with the global configure right see and may click the "Reset quality gate" button.

Useres that may configure the job/build should also be able to reset the quality gate.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: plugin/src/test/java/io/jenkins/plugins/analysis/core/model/ResetQualityGateCommandTest.java [hunk 1]
```diff
@@ -134,12 +145,13 @@ private ResultAction createResultAction(final QualityGateStatus status, final St
     }
 
     private FreeStyleBuild attachReferenceBuild(final boolean hasNoReferenceBuild,
-            final ResultAction resultAction) {
+            final boolean hasRight, final ResultAction resultAction) {
```
Comment (uhafner) [inline plugin/src/test/java/io/jenkins/plugins/analysis/core/model/ResetQualityGateCommandTest.java:148]:
```suggestion
            final boolean hasConfigurePermission, final ResultAction resultAction) {
```

File: plugin/src/test/java/io/jenkins/plugins/analysis/core/model/ResetQualityGateCommandTest.java [hunk 2]
```diff
@@ -134,12 +145,13 @@ private ResultAction createResultAction(final QualityGateStatus status, final St
...
         FreeStyleBuild selectedBuild = mock(FreeStyleBuild.class);
         when(selectedBuild.getActions(ResetReferenceAction.class)).thenReturn(
                 Lists.list(new ResetReferenceAction(hasNoReferenceBuild ? "other" : ID)));
         when(selectedBuild.getActions(ResultAction.class)).thenReturn(Lists.list(resultAction));
         when(selectedBuild.getAction(ResultAction.class)).thenReturn(resultAction);
+        when(selectedBuild.hasPermission(Item.CONFIGURE)).thenReturn(hasRight);
```
Comment (uhafner) [inline plugin/src/test/java/io/jenkins/plugins/analysis/core/model/ResetQualityGateCommandTest.java:154]:
```suggestion
        when(selectedBuild.hasPermission(Item.CONFIGURE)).thenReturn(hasConfigurePermission);
```

**Review by uhafner — APPROVED** [review]
Thanks for your PR!

**Comment by nitschSB** [comment]
hi, i think Ticket https://issues.jenkins.io/browse/JENKINS-70172 may be closed.

are there any plans for a release?

**Comment by uhafner** [comment]
Thanks for the reminder, I forgot to create a new release... it is on its way now.

**Comment by mawinter69** [comment]
This can't work. A run is not an Item that can be configured. 
`selectedBuild.getParent().hasPermission(Item.CONFIGURE)` should work.
But maybe it is better to even have a dedicated permission for runs to reset the quality gate. I could imagine that people should be allowed to reset the quality gate without them having the need to configure jobs.

**Pull request merged**